darwin/exporter/formats/mask.py

```diff
@@ -62,46 +394,60 @@ def export(annotation_files: Iterable[dt.AnnotationFile], output_dir: Path, mode
 
-        mask: Image.Image = np.zeros((height, width)).astype(np.uint8)
-        annotations = [a for a in annotation_file.annotations if ispolygon(a.annotation_class)]
-        for a in annotations:
-            if isinstance(a, dt.VideoAnnotation):
-                print(f"Skipping video annotation from file {annotation_file.filename}")
-                continue
+        mask: NDArray = np.zeros((height, width)).astype(np.uint8)
+        annotations: List[dt.AnnotationLike] = [
+            a
+            for a in annotation_file.annotations
+            if a.annotation_class.annotation_type in ["polygon", "complex_polygon", "raster_layer", "mask"]
+        ]
 
-            cat = a.annotation_class.name
-            if a.annotation_class.annotation_type == "polygon":
-                polygon = a.data["path"]
-            elif a.annotation_class.annotation_type == "complex_polygon":
-                polygon = a.data["paths"]
-            sequence = convert_polygons_to_sequences(polygon, height=height, width=width)
-            draw_polygon(mask, sequence, palette[cat])
+        render_type = get_render_mode(annotations)
 
+        if render_type == "raster":
+            # Add categories to list
+            errors, mask, categories, colours = render_raster(
+                mask, colours, categories, annotations, annotation_file, height, width
+            )
+
+        else:
+            #  Add categories to list
+            errors, mask, categories, colours = render_polygons(
+                mask, colours, categories, annotations, annotation_file, height, width
+            )
+
+        if errors:
+            print(f"Errors rendering {annotation_file.filename}:")
+            for e in errors:
+                print(e)
+
+            raise DarwinException.from_multiple_exceptions(errors)
+
+        # Map to palette
+        mask = np.array(mask, dtype=np.uint8)  # Final double check that type is using correct dtype
+
+        palette = get_palette(mode, categories)
         if mode == "rgb":
-            mask = Image.fromarray(mask, "P")
-            mask.putpalette(RGB_colors)
+            rgb_colours, palette_rgb = get_rgb_colours(categories)
+            image = Image.fromarray(mask, "P")
+            image.putpalette(rgb_colours)
+            image = image.convert("RGB")
+        elif mode == "grey":
+            for value, colour in enumerate(palette.values()):
+                mask = np.where(mask == value, colour, mask)
+            image = Image.fromarray(mask)
         else:
-            mask = Image.fromarray(mask)
-        mask.save(outfile)
+            image = Image.fromarray(mask)
+        image.save(outfile)
 
     with open(output_dir / "class_mapping.csv", "w") as f:
-        f.write(f"class_name,class_color\n")
+        writer = csv_writer(f)
+        writer.writerow(["class_name", "class_color"])
+
+        if "__background__" in categories:
```

Comment:
@almazan I've added this, but do CSV files actually need the `__background__` class included for any reason?

Reply:
Have confirmed with Jon how he'd recommend this behave